### PR TITLE
Add ClinicalTrialsGovCondition model

### PR DESCRIPTION
**Pull request by zhx828**
This fixes https://github.com/oncokb/oncokb/issues/3404

### List view
Screen Shot 2023-02-27 at 5 33 40 PM

### Create/Update view
Screen Shot 2023-02-27 at 5 35 31 PM

### Read view
Screen Shot 2023-02-27 at 5 34 54 PM

### CODE REVIEW

**Comment by zhx828** [comment]
@calvinlu3 mostly boiler code, nothing was changed on the backend. In the frontend, I modified how the cancer type is associated with ct.gov condition.

File: src/main/webapp/app/shared/select/CancerTypeSelect.tsx [hunk 1]
```diff
@@ -8,11 +8,12 @@ import { IRootStore } from 'app/stores/createStore';
 import { connect } from '../util/typed-inject';
 import { ICancerType } from '../model/cancer-type.model';
 import { ITEMS_PER_PAGE } from '../util/pagination.constants';
+import { getCancerTypeName } from 'app/shared/util/utils';
 
 interface ICancerTypeSelectProps extends SelectProps, StoreProps {}
 
 const getAllMainTypes = (cancerTypeList: ICancerType[]) => {
-  return _.uniq(cancerTypeList.filter(cancerType => cancerType.level >= 0)).sort();
+  return _.uniq(cancerTypeList.filter(cancerType => cancerType.level <= 0)).sort();
```
Comment (scarrero4660) [inline src/main/webapp/app/shared/select/CancerTypeSelect.tsx:16]: Why is this filtering by checking if levels <= 0? Wouldn't this remove all of the cancer types?

File: src/main/webapp/app/entities/clinical-trials-gov-condition/clinical-trials-gov-condition-detail.tsx [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+export const ClinicalTrialsGovConditionDetail = (props: IClinicalTrialsGovConditionDetailProps) => {
+  useEffect(() => {
+    props.getEntity(props.match.params.id);
+  }, []);
+
+  const clinicalTrialsGovConditionEntity = props.clinicalTrialsGovConditionEntity;
+  return (
+    <Row>
+      <Col md="8">
+        <h2 data-cy="clinicalTrialsGovConditionDetailsHeading">CT.gov Condition</h2>
```
Comment (scarrero4660) [inline src/main/webapp/app/entities/clinical-trials-gov-condition/clinical-trials-gov-condition-detail.tsx:22]: Perhaps it is better to use full "ClinicalTrials.gov" in the heading here

File: src/main/webapp/app/shared/select/CancerTypeSelect.tsx (hunk 1 above)
Comment (zhx828) [reply src/main/webapp/app/shared/select/CancerTypeSelect.tsx:16]: @scarrero4660 within the list of cancer types, we have concepts `main type` and `subtype`. They are the concepts from OncoTree. They are also called `Cancer Type` and `Cancer Type Details`. Any level 0 are main type, and any level >0 are subtype. OncoKB also adds additional cancer types for our purpose, such as `All Solid Tumors`, for these, we use level -1.

File: src/main/webapp/app/entities/clinical-trials-gov-condition/clinical-trials-gov-condition-detail.tsx (hunk 2 above)
Comment (zhx828) [reply src/main/webapp/app/entities/clinical-trials-gov-condition/clinical-trials-gov-condition-detail.tsx:22]: I want to keep it short. CT.gov is a well perceived usage.

**Review by calvinlu3 — APPROVED** [review]

**Pull request merged**